doozer/doozerlib/cli/scan_osh.py

```diff
@@ -1,29 +1,240 @@
         self.error_nvrs = []
+        self.version = self.runtime.group.split("-")[-1]
+        self.jira_project = "OCPBUGS"
 
         # Initialize runtime and brewhub session
         self.runtime.initialize(clone_distgits=False)
         self.koji_session = koji.ClientSession(self.runtime.group_config.urls.brewhub)
 
+        # Initialize JIRA client
+        self.jira_client: JIRA = self.runtime.build_jira_client()
+
+    def get_latest_scan_task_id(self, brew_package_name: str):
+        """
+        Find the last successful scan for a particular package and retrieve its ID
+        """
+        # Eg: osh-cli find-tasks --regex ose-network-tools-container-v4.15
+        cmd = f"osh-cli find-tasks --regex {brew_package_name}"
+
+        _, result, _ = cmd_gather(cmd)
+
+        finished_tasks = result.strip().split("\n")
+
+        for task_id in finished_tasks:
+            cmd = f"osh-cli task-info {task_id}"
+            self.runtime.logger.info(f"Running: {cmd}")
+
+            _, result, _ = cmd_gather(cmd)
+
+            # Check if the task finished successfully
+            if "state_label = CLOSED" in result:
+                return task_id
+
+    @staticmethod
+    def check_if_vulnerabilities_exist(task_id: str, nvr: str) -> bool:
+        """
+        ProdSec only wants us to check the scan-results-imp.js file of a scan and see if vulnerabilities are reported.
+        """
+
+        response = requests.get(SCAN_RESULTS_URL_TEMPLATE.format(task_id=task_id, nvr=nvr))
+
+        if len(response.json()["defects"]) > 0:
+            return True
+
+        return False
+
+    def get_distgit_name_from_brew_nvr(self, nvr: str) -> str:
+        """
+            Returns the distgit name from the brew nvr
+
+            :param nvr: Full NVR eg: openshift-enterprise-pod-container-v4.14.0-202310031045.p0.g9ef6de6.assembly.stream
+            """
+        brew_package_name = self.koji_session.getBuild(buildInfo=nvr, strict=True)["package_name"]
+
+        _, output, _ = cmd_gather(f"doozer --disable-gssapi -g {self.runtime.group} "
+                                  f"images:print --short '{{component}}: {{name}}'")
+        result = []
+        for line in output.splitlines():
+            result.append(line.split(": "))
+
+        dict_data = {}
+        for line in result:
+            if line:
+                dict_data[line[0]] = line[1]
+
+        return dict_data[brew_package_name]
+
+    def create_update_ocpbugs_ticket(self, packages: dict):
+        """
+        Check if an OCPBUGS ticket already exists for a particular package. If it doesn't create one and update
+        the contents to point to the latest scan results
+        """
+        for brew_package_name in packages:
+            data = packages[brew_package_name]
+            distgit_name = self.get_distgit_name_from_brew_nvr(data["nvr"])
+
+            image_meta: ImageMetadata = self.runtime.image_map[distgit_name]
```

Comment:
Suggesting we allow jira ticket creation to be enabled/disabled at the metadata level. We can default it to disabled at first and onboard a single image/rpm and to how things work. e.g. in the metadata, require
`scanning.jira_integration.enabled: true` to be set explicitly before opening issues. 
We will also probably want something similar in group.yml to enable Jira integration slowly across z-streams so as to not overwhelm folks.

Reply:
Added in https://github.com/openshift-eng/art-tools/pull/141/commits/b7efbdde56badb6ce9bd82aab0ba3ace9daa0e35